src/coprocessor/tracker.rs

```diff
@@ -150,27 +150,32 @@ impl Tracker {
     /// Get current item's ExecDetail according to previous collected metrics.
     /// TiDB asks for ExecDetail to be printed in its log.
     /// WARN: TRY BEST NOT TO USE THIS FUNCTION.
-    pub fn get_item_exec_details(&self) -> kvrpcpb::ExecDetails {
+    pub fn get_item_exec_details(&self) -> (kvrpcpb::ExecDetails, kvrpcpb::ExecDetailsV2) {
         assert_eq!(self.current_stage, TrackerState::ItemFinished);
         self.exec_details(self.item_process_time)
     }
 
     /// Get ExecDetail according to previous collected metrics.
     /// TiDB asks for ExecDetail to be printed in its log.
-    pub fn get_exec_details(&self) -> kvrpcpb::ExecDetails {
+    pub fn get_exec_details(&self) -> (kvrpcpb::ExecDetails, kvrpcpb::ExecDetailsV2) {
         assert_eq!(self.current_stage, TrackerState::ItemFinished);
         self.exec_details(self.total_process_time)
     }
 
-    fn exec_details(&self, measure: Duration) -> kvrpcpb::ExecDetails {
+    fn exec_details(&self, measure: Duration) -> (kvrpcpb::ExecDetails, kvrpcpb::ExecDetailsV2) {
+        // For compatibility, ExecDetails field is still filled.
         let mut exec_details = kvrpcpb::ExecDetails::default();
 
-        let mut handle = kvrpcpb::HandleTime::default();
-        handle.set_process_ms((time::duration_to_sec(measure) * 1000.0) as i64);
-        handle.set_wait_ms((time::duration_to_sec(self.wait_time) * 1000.0) as i64);
-        exec_details.set_handle_time(handle);
+        let mut td = kvrpcpb::TimeDetail::default();
+        td.set_process_wall_time_ms((time::duration_to_sec(measure) * 1000.0) as i64);
```

Comment:
Why is the time precision second? There exists `duration_to_ms`.

Reply:
Good catch